File: medium/2_add_two_numbers.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func addTwoNumbers(l1 *ListNode, l2 *ListNode) *ListNode {
    result := &ListNode{}
    temp := result
    for l1 != nil || l2 != nil {
        if l1 != nil {
            temp.Val += l1.Val
            l1 = l1.Next
        }
        if l2 != nil {
            temp.Val += l2.Val
            l2 = l2.Next
        }
        if temp.Val > 9 {
            temp.Next = &ListNode{Val: temp.Val / 10}
            temp.Val %= 10
        } else if l1 != nil || l2 != nil {
            temp.Next = &ListNode{}
        }
        temp = temp.Next
    }
    return result
}

File: medium/2_add_two_numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        result = ListNode(0)
        temp = result
        while l1 or l2:
            if l1:
                temp.val += l1.val
                l1 = l1.next
            if l2:
                temp.val += l2.val
                l2 = l2.next

            if temp.val > 9:
                temp.next = ListNode(val=(temp.val // 10))
                temp.val %= 10
            elif l1 or l2:
                temp.next = ListNode()
            temp = temp.next

        return result

